### PR TITLE
PLAT-565 Set systemUser as currentUser before executing programs

### PR DESCRIPTION
**Pull request by thees-k**
Test plan:

To reproduce the error add to `BufferedEmailSendProgram` in line 21:

```
System.out.println(CurrentUser.get().getLoginName());
```

When you start `BufferedEmailSendProgram` you will see that the login name of the logged-user is printed on console.

After the patch "system.user" will be printed out instead.

### CODE REVIEW

File: platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramStarter.java [hunk 1]
```diff
@@ -53,5 +55,6 @@ private void prepareExecution() {
 
 		ISOCalendar.setDefaultTimeZone();
 		LogDb.setProcessClass(program.getClass());
+		CurrentUser.set(AGUser.getSystemUser());
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramStarter.java:58]: I think we should use `CurrentUserScope` in the surrounding `try` block instead. E.g.
```
try (var scope = new CurrentUserScope(AGUser.getSystemUser())) {
	prepareExecution();
	program.executeProgram();
}
```
That would prevent problems when `start()` gets called directly from a UI thread.

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramStarter.java (hunk 1 above)
Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramStarter.java:58]: I think the original solution was fine, considering the already existing code.

Comment (alexander-schmidt) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramStarter.java:58]: I disagree. The proposed change eliminates a side effect which couly very easily be triggered by future code, while not really increasing complexity.

**Comment by thees-k** [comment]
In my eyes both solutions show clearly why executor of a program is sytemUser.

But I don't care so I adjusted the code.

File: platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramStarter.java [hunk 2]
```diff
@@ -40,12 +42,9 @@ public static void main(String[] arguments) throws Throwable {
 
 	public void start() {
 
-		try {
+		try (var scope = new CurrentUserScope(AGUser.getSystemUser())) {
 			prepareExecution();
 			program.executeProgram();
-		} catch (Exception exception) {
```
Comment (oliver-richers) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramStarter.java:46]: i don't think this `catch` should not be removed

**Review by oliver-richers — CHANGES_REQUESTED** [review]

File: platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramStarter.java (hunk 2 above)
Comment (thees-k) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/program/ProgramStarter.java:46]: I restored the catch block

**Review by oliver-richers — APPROVED** [review]
looks fine now

**Pull request merged**